### PR TITLE
Disable max money check on networks with no supply cap

### PR DESCRIPTION
**Pull request by rnicoll**
More altcoin support stuff

### CODE REVIEW

File: core/src/main/java/org/bitcoinj/core/Transaction.java [hunk 1]
```diff
@@ -1256,7 +1256,7 @@ public void verify() throws VerificationException {
                     throw new VerificationException.NegativeValueOutput();
                 valueOut = valueOut.add(output.getValue());
                 // Duplicate the MAX_MONEY check from Coin.add() in case someone accidentally removes it.
```
Comment (schildbach) [inline core/src/main/java/org/bitcoinj/core/Transaction.java]: This comment can go, too.

**Comment by schildbach** [comment]
There is some more of these checks in the code, maybe you want to tackle more of them at the same time?

**Comment by rnicoll** [comment]
I've now replaced everything except the case where it's used to send all from a wallet. I'll do that one too, later, but for now I'd rather give this time to settle first.

**Comment by schildbach** [comment]
I just discovered that some of the checks you're replacing in this PR are actually off-by-one.

Assuming that my fixes in PR 1177 are correct, do you think you could rebase this PR on PR 1177 and fix the checks in your replacement too?

**Comment by rnicoll** [comment]
#1177 looks good, and I've rebased this now

**Comment by schildbach** [comment]
Merged, thanks!

**Pull request merged**